resources/asciidoctor/lib/open_in_widget/extension.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+
+require 'fileutils'
+
+require_relative '../scaffold.rb'
+
+##
+# Extensions for enriching certain source blocks with "OPEN IN CONSOLE",
+# "OPEN IN SENSE", "OPEN IN KIBANA", AND/OR "COPY_AS_CURL".
+#
+# Usage
+#
+#   [source,console]
+#   ---------
+#   GET /
+#   ---------
+#
+# or
+#
+#   [source,sense]
+#   ---------
+#   GET /
+#   ---------
+#
+# or
+#
+#   [source,kibana]
+#   ---------
+#   GET /
+#   ---------
+#
+# or
+#
+#   [source,sense,snippet=path/to/snippet.console]
+#   ---------
+#   GET /
+#   ---------
+#
+class OpenInWidget < TreeProcessorScaffold
+  include Asciidoctor::Logging
+
+  CALLOUT_SCAN_RX = / ?#{Asciidoctor::CalloutScanRx}/.freeze
+
+  def process_block(block)
+    return unless block.context == :listing && block.style == 'source'
+
+    lang = block.attr 'language'
+    return unless %w[console sense kibana].include? lang
```

Comment:
I would put the following in a `unless %w[console sense kibana].include? lang` condition.